### ISSUE TITLE
Auto renewal Changes

### ISSUE THREAD

**Issue by bhanu2217** [faveosuite/faveo-invoicing-community#2566]
Auto renewal button should be enabled under following circumstances:

- Payment gateway response tell your that card data was captured
- Don't enable auto renewal without taking customer card details
- Payment gateway should confirm that customer card details are captured only then auto renewal should be enabled
- Payment gateway log should show how many customers enabled auto renewal
- Payment gateway lof search/filter show be able to search for customers who's auto renewal is currently active or working
- If auto renewal is failing it should become disabled and customer has to enable it again

**QA Touch cases** (15) — project `ekXp`, module **orders**

TR0983, TR0984, TR0985, TR0986, TR0987, TR0988, TR0989, TR0990, TR0991, TR0992, TR0993, TR0994, TR0995, TR0996, TR0997

**Comment by faveobot** [comment]
## Test cases for review

**15 case(s)** authored for this enhancement and created in QA Touch module **orders** (project `ekXp`).

> [!IMPORTANT]
> **Some content on this issue could not be opened, so these cases do not cover it.**
>
> - `unanalysed` "If auto renewal is failing it should become disabled" — the auto-disable happens in ConcretePostSubscriptionHandleController::disableAutorenewalStatusByOrderId, reached only from the failed-renewal-payment mail path (line 272), i.e. the subscription renewal cron/webhook; there is no UI action a tester can click to trigger a failing renewal, so that case carries a precondition the executor may not be able to establish.
> - `unanalysed` "Payment gateway log should show how many customers enabled auto renewal" — no counter or auto-renewal column exists today in resources/views/themes/default1/common/payment-log.blade.php (columns are Date, User, Order No, Amount, Description, Payment Method, Status), so the two payment-log cases below assert the requested new behaviour rather than something already present.
> - `unanalysed` The issue gives no wording for the requested counter/filter labels, so those two cases assert the presence and correctness of the affordance without pinning an exact label.
>
> Paste the relevant detail into the issue (or attach it as an image or PDF) and re-apply `QA: Test case needed` to add the missing cases.

| Code | Discipline | Title | Steps |
|---|---|---|---|
| `TR0983` | functional | Customer enables auto renewal only after Stripe captures card details | 6 |
| `TR0984` | functional | Auto renewal cannot be switched on while the global setting is disabled | 5 |
| `TR0985` | ux | Auto renewal modal blocks card entry until a gateway is chosen | 5 |
| `TR0986` | functional | Declined card leaves auto renewal inactive | 5 |
| `TR0987` | functional | Customer disables auto renewal and must re-capture a card to enable it again | 5 |
| `TR0988` | functional | Admin cannot switch on auto renewal from the order page | 5 |
| `TR0989` | regression | Payment log records the gateway confirmation for each auto renewal enable | 5 |
| `TR0990` | functional | Payment log shows how many customers have auto renewal enabled | 5 |
| `TR0991` | functional | Payment log can be filtered to customers whose auto renewal is currently active | 5 |
| `TR0992` | functional | Admin toggles the instance-wide Auto Renewal setting and it persists | 5 |
| `TR0993` | security | Customer cannot change auto renewal on an order they do not own | 5 |
| `TR0994` | security | Guest cannot reach the auto renewal endpoints | 5 |
| `TR0995` | security | Customer cannot reach the payment log or the global Auto Renewal setting | 5 |
| `TR0996` | api | Disable auto renewal JSON endpoint contract and CSRF handling | 5 |
| `TR0997` | regression | Failed auto renewal payment leaves auto renewal disabled until the customer re-enables it | 5 |

<details><summary><code>TR0983</code> Customer enables auto renewal only after Stripe captures card details</summary>

**Precondition:** Stripe plugin is active with working test API keys (admin > Settings). The global Auto Renewal toggle on Settings > System is ON. The customer account has at least one order under My Orders whose product has an active renewal plan price — the switch is rendered disabled when no plan price exists (show-order.blade.php:740-743).

| # | Step | Expected result |
|---|---|---|
| 1 | Log in at /login with the QA_CLIENT_ customer credentials and open My Orders, then click the order number to open /my-order/{id}. | The order detail page opens and a tab labelled "Auto Renewal" is present in the tab strip. |
| 2 | Click the "Auto Renewal" tab. | The Auto Renewal panel shows the label "Auto Renewal:" with a toggle switch that is off, and "Status:" showing "Inactive" in red. No "Payment Gateway" or "Subscription Enabled Date" row is shown. |
| 3 | Click the Auto Renewal toggle switch to turn it on. | A modal titled "Auto Renewal" opens with the required dropdown "Select the payment gateway" (default option "Choose your option") and buttons "Close" and "Save". |
| 4 | Select "Stripe" in the "Select the payment gateway" dropdown and click "Save". | The Auto Renewal modal closes and a modal titled "Enter card details" opens with Card Number, Expiry Date and CVC fields, a Total line showing the 1-unit verification amount, and a "PAY NOW" button. |
| 5 | Enter card number 4242 4242 4242 4242, expiry 12/34, CVC 123 and click "PAY NOW". | The card modal closes and a green success alert appears reading "Your Card details are updated successfully."; the page then reloads. |
| 6 | After the reload, open the "Auto Renewal" tab again. | The toggle is on, "Status:" shows "Active" in green, and rows "Payment Gateway" (value stripe) and "Subscription Enabled Date:" (today's date) are now displayed. |

</details>

<details><summary><code>TR0984</code> Auto renewal cannot be switched on while the global setting is disabled</summary>

**Precondition:** The customer account has an order with an active renewal plan price and auto renewal currently Inactive.

| # | Step | Expected result |
|---|---|---|
| 1 | Log in at /login as the QA_ADMIN_ admin, open /settings/system, and switch the "Auto Renewal" toggle (the row with the question-mark tooltip "Enable or disable auto-renewal for all users.") to off, then click "Update". | The page reloads showing the flash message "Updated Successfully" and the "Auto Renewal" toggle is off. |
| 2 | Log out, log in at /login as the QA_CLIENT_ customer, open My Orders and open the order via /my-order/{id}, then click the "Auto Renewal" tab. | The Auto Renewal panel is shown with the toggle off and "Status:" reading "Inactive". |
| 3 | Click the Auto Renewal toggle switch to turn it on. | A dialog titled "Info" appears with the text "Auto-renewal is currently not available. Please contact the administrator for assistance." and an "OK" button. No gateway-selection modal and no card-details modal open. |
| 4 | Click "OK" in the dialog. | The dialog closes and the Auto Renewal toggle is back in the off position. |
| 5 | Reload /my-order/{id} and open the "Auto Renewal" tab. | The toggle is still off and "Status:" still reads "Inactive"; no "Payment Gateway" or "Subscription Enabled Date" row appears. |

</details>

<details><summary><code>TR0985</code> Auto renewal modal blocks card entry until a gateway is chosen</summary>

**Precondition:** Global Auto Renewal setting is ON. The customer has an order with an active renewal plan price and auto renewal Inactive.

| # | Step | Expected result |
|---|---|---|
| 1 | Log in as the QA_CLIENT_ customer, open /my-order/{id} for the order and click the "Auto Renewal" tab. | The Auto Renewal panel is displayed with the toggle off and "Status:" "Inactive". |
| 2 | Click the Auto Renewal toggle switch to turn it on. | The "Auto Renewal" modal opens with the "Select the payment gateway" dropdown showing "Choose your option" and a red asterisk beside the label. |
| 3 | Without choosing a gateway, click "Save". | The modal stays open, the message "Please Select the Payment" is shown in red under/next to the dropdown, the dropdown border turns red and the button label changes to "Validate". No card-details modal opens. |
| 4 | Select "Stripe" from the dropdown and click the button again. | The error text is no longer blocking; the "Enter card details" modal opens with Card Number, Expiry Date and CVC fields. |
| 5 | Close the card-details modal with the × button without entering a card, then reload /my-order/{id} and open the "Auto Renewal" tab. | "Status:" still reads "Inactive" and the toggle is off — no auto renewal was enabled without a captured card. |

</details>

<details><summary><code>TR0986</code> Declined card leaves auto renewal inactive</summary>

**Precondition:** Stripe plugin active with test API keys. Global Auto Renewal setting is ON. The customer has an order with an active renewal plan price and auto renewal Inactive.

| # | Step | Expected result |
|---|---|---|
| 1 | Log in as the QA_CLIENT_ customer, open /my-order/{id} and click the "Auto Renewal" tab. | The panel shows the toggle off and "Status:" "Inactive". |
| 2 | Turn the toggle on, select "Stripe" in "Select the payment gateway" and click "Save". | The "Enter card details" modal opens with the Card Number, Expiry Date and CVC fields. |
| 3 | Enter the Stripe decline test card 4000 0000 0000 0002, expiry 12/34, CVC 123 and click "PAY NOW". | A red error alert is displayed reading "Something went wrong. Try with a different payment method." (prefixed by the "Oh snap!" heading); no success message appears. |
| 4 | Reload /my-order/{id} and open the "Auto Renewal" tab. | The toggle is off and "Status:" reads "Inactive" in red; no "Payment Gateway" or "Subscription Enabled Date" row is present. |
| 5 | Log out, log in as the QA_ADMIN_ admin and open /settings/paymentlog. | The Payment Log table contains a row for the customer with Description "Payment method updated", Payment Method "Stripe" and Status shown as the "Failed" link — the failed capture is recorded and auto renewal was not enabled. |

</details>

<details><summary><code>TR0987</code> Customer disables auto renewal and must re-capture a card to enable it again</summary>

**Precondition:** The customer has an order whose Auto Renewal status is "Active" (run the "Customer enables auto renewal only after Stripe captures card details" case first). Global Auto Renewal setting is ON.

| # | Step | Expected result |
|---|---|---|
| 1 | Log in as the QA_CLIENT_ customer, open /my-order/{id} and click the "Auto Renewal" tab. | The toggle is on and "Status:" reads "Active" in green, with "Payment Gateway" and "Subscription Enabled Date:" rows shown. |
| 2 | Click the toggle switch to turn auto renewal off. | A green success alert appears reading "Auto subscription Disabled successfully." and the page reloads a few seconds later. |
| 3 | After the reload, open the "Auto Renewal" tab. | The toggle is off, "Status:" reads "Inactive" in red and the "Payment Gateway" / "Subscription Enabled Date:" rows are gone. |
| 4 | Click the toggle switch to turn auto renewal on again. | The "Auto Renewal" gateway-selection modal opens again — the status does not flip to Active without a new card capture. |
| 5 | Select "Stripe", click "Save", enter card 4242 4242 4242 4242 / 12/34 / 123 and click "PAY NOW", then reopen the "Auto Renewal" tab after the reload. | The success alert "Your Card details are updated successfully." is shown and after reload "Status:" reads "Active" with "Payment Gateway" and "Subscription Enabled Date:" populated. |

</details>

<details><summary><code>TR0988</code> Admin cannot switch on auto renewal from the order page</summary>

**Precondition:** At least one order exists for the customer account whose Auto Renewal status is Inactive (a freshly created order satisfies this).

| # | Step | Expected result |
|---|---|---|
| 1 | Log in at /login as the QA_ADMIN_ admin and open the customer's order from the Orders list (/orders/{id}). | The order detail page opens showing the order number and the customer's name. |
| 2 | Expand the "Auto Renewal" section of the page. | The section shows a row "Auto Renewal for Future Subscription" with a toggle switch and a row "Status:" reading "Inactive" in red. |
| 3 | Click the "Auto Renewal for Future Subscription" toggle switch. | The switch does not change state (it is disabled); no gateway-selection or card-details modal opens and no success message appears. |
| 4 | Reload /orders/{id} and expand the "Auto Renewal" section again. | "Status:" still reads "Inactive" and no "Payment Method" / "Subscription start date" rows are shown — no auto renewal was enabled without card capture. |
| 5 | Log out, log in as the QA_CLIENT_ customer and open /my-order/{id} for the same order, then click the "Auto Renewal" tab. | The client-side "Status:" also reads "Inactive", confirming nothing was activated by the admin's click. |

</details>

<details><summary><code>TR0989</code> Payment log records the gateway confirmation for each auto renewal enable</summary>

**Precondition:** The customer has just enabled auto renewal on an order with a successful card capture (run the primary enable case first). Note the order number.

| # | Step | Expected result |
|---|---|---|
| 1 | Log in at /login as the QA_ADMIN_ admin and open /settings/paymentlog. | The "Payment Log" page loads with a From/Till search panel and a table whose headers are Date, User, Order No, Amount, Description, Payment Method and Status. |
| 2 | Type the customer's name or the order number into the table's Search box. | The table filters down to rows for that customer/order, including one row whose Description is "Payment method updated". |
| 3 | Read that row's Payment Method, Status and Date cells. | Payment Method reads "Stripe", Status indicates success (not the red "Failed" link) and Date is the date/time the customer enabled auto renewal. |
| 4 | Click the order number link in the Order No column of that row. | The admin order detail page for that order opens, and its "Auto Renewal" section shows "Status: Active" with the "Payment Method" row. |
| 5 | Go back to /settings/paymentlog, set "From" and "Till" to a past date range that excludes today and click "Search". | The auto-renewal "Payment method updated" row is no longer listed; clicking "Reset" brings the full log back including that row. |

</details>

<details><summary><code>TR0990</code> Payment log shows how many customers have auto renewal enabled</summary>

**Precondition:** Exactly one customer order currently has Auto Renewal status "Active" (enable it via the client Auto Renewal tab first) and the order number is known.

| # | Step | Expected result |
|---|---|---|
| 1 | Log in at /login as the QA_ADMIN_ admin and open /settings/paymentlog. | The "Payment Log" page loads and displays a count of customers with auto renewal currently enabled. |
| 2 | Read the displayed count. | The count reads 1, matching the single order whose auto renewal is active. |
| 3 | In another tab, log in as the QA_CLIENT_ customer, open /my-order/{id} for that order, click the "Auto Renewal" tab and switch the toggle off. | The success alert "Auto subscription Disabled successfully." appears and after reload "Status:" reads "Inactive". |
| 4 | Return to the admin tab and reload /settings/paymentlog. | The auto-renewal count now reads 0. |
| 5 | As the customer, re-enable auto renewal on the order with card 4242 4242 4242 4242 / 12/34 / 123, then reload /settings/paymentlog as the admin. | The count reads 1 again — the figure counts customers whose auto renewal is currently enabled, not the number of log rows. |

</details>

<details><summary><code>TR0991</code> Payment log can be filtered to customers whose auto renewal is currently active</summary>

**Precondition:** One customer order has Auto Renewal status "Active" and at least one other order/customer in the payment log has auto renewal inactive (e.g. an order where a card capture previously failed or was later disabled).

| # | Step | Expected result |
|---|---|---|
| 1 | Log in at /login as the QA_ADMIN_ admin and open /settings/paymentlog. | The "Payment Log" page loads with the From/Till search panel, "Search" and "Reset" buttons and the log table. |
| 2 | Apply the filter for customers whose auto renewal is currently active and click "Search". | The table reloads showing only rows for the customer whose auto renewal is active; the customer with auto renewal inactive is not listed. |
| 3 | As the customer (separate browser/session), open the active order's "Auto Renewal" tab and switch auto renewal off. | The alert "Auto subscription Disabled successfully." appears and "Status:" becomes "Inactive". |
| 4 | Back as admin, reapply the active-auto-renewal filter on /settings/paymentlog and click "Search". | No rows are returned for that customer; the table shows the empty/no-matching-records message. |
| 5 | Click "Reset". | The filter clears and the full unfiltered payment log is listed again, including the rows for the customer who just disabled auto renewal. |

</details>

<details><summary><code>TR0992</code> Admin toggles the instance-wide Auto Renewal setting and it persists</summary>

**Precondition:** The System settings form's existing required values (Company, Company email, Website, Phone, Address, State, Country, Default currency — see app/Http/Requests/Common/SettingsRequest.php) are already populated; do not clear them.

| # | Step | Expected result |
|---|---|---|
| 1 | Log in at /login as the QA_ADMIN_ admin and open /settings/system. | The System settings page loads and contains a row labelled "Auto Renewal" with a toggle switch and a question-mark tooltip reading "Enable or disable auto-renewal for all users." |
| 2 | Note the toggle's current position, switch it to ON and click "Update" without changing any other field. | The page reloads with the flash message "Updated Successfully" and the "Auto Renewal" toggle is shown ON. |
| 3 | Reload /settings/system. | The "Auto Renewal" toggle is still ON after the reload — the setting persisted. |
| 4 | Switch the toggle to OFF and click "Update", then reload /settings/system. | The flash "Updated Successfully" appears and the toggle stays OFF after the reload. |
| 5 | Restore the toggle to its original position from step 2 and click "Update". | "Updated Successfully" is shown and the toggle is back in its original position. |

</details>

<details><summary><code>TR0993</code> Customer cannot change auto renewal on an order they do not own</summary>

**Precondition:** An order that belongs to the QA_ADMIN_ account (as a purchaser, not as an admin) and has a subscription row — establish it by buying a product while logged in as the admin account, and note its order id. Its id must differ from any order owned by the QA_CLIENT_ customer.

| # | Step | Expected result |
|---|---|---|
| 1 | Log in at /login as the QA_CLIENT_ customer in a browser and confirm the session by loading /client-dashboard. | The client dashboard renders for the customer account. |
| 2 | From that same authenticated session, send POST /renewal-disable with form field order_id set to the order id owned by the other account, including the page's CSRF token. | HTTP 403 with JSON body {"success":false,"message":"You do not have the necessary permissions to perform this action."} |
| 3 | From the same session, send POST /strRenewal-enable with order_id set to that same foreign order id and any stripeToken value, plus the CSRF token. | HTTP 403 with JSON body {"success":false,"message":"You do not have the necessary permissions to perform this action."} — no card capture is attempted. |
| 4 | Log out and log in as the QA_ADMIN_ account, open /my-order/{id} for that order and click the "Auto Renewal" tab. | The auto renewal Status is unchanged by the customer's two requests (still whatever it was before the attempts). |
| 5 | Open /my-order/{id} of the foreign order id directly while logged in as the QA_CLIENT_ customer. | The customer is not shown another account's order detail — the request is refused or redirected away with an error instead of rendering the Auto Renewal tab for that order. |

</details>

<details><summary><code>TR0994</code> Guest cannot reach the auto renewal endpoints</summary>

**Precondition:** No active session (fresh browser profile or cleared cookies). An order id that exists in the instance is known.

| # | Step | Expected result |
|---|---|---|
| 1 | With no session, open /my-order/1 (or another known order id) in the browser. | The order page does not render; the browser lands on the login page. |
| 2 | With no session, send POST /renewal-disable with order_id set to that order id. | The response is a 302 redirect to /login (or the login page HTML); no JSON success body and no 200 with a success message. |
| 3 | With no session, send POST /strRenewal-enable with order_id and a stripeToken value. | The response is a 302 redirect to /login; nothing is enabled. |
| 4 | With no session, send POST /rzpRenewal-disable/{orderid} with a razorpay_payment_id value. | The response is a 302 redirect to /login. |
| 5 | Log in as the QA_CLIENT_ customer, open /my-order/{id} for their own order and check the "Auto Renewal" tab. | The auto renewal Status is unchanged by the four anonymous requests. |

</details>

<details><summary><code>TR0995</code> Customer cannot reach the payment log or the global Auto Renewal setting</summary>

**Precondition:** The payment log contains at least one "Payment method updated" row (i.e. some customer enabled auto renewal at least once).

| # | Step | Expected result |
|---|---|---|
| 1 | Log in at /login as the QA_CLIENT_ customer and confirm the session by loading /client-dashboard. | The client dashboard renders. |
| 2 | Navigate directly to /settings/paymentlog. | The Payment Log page is not shown; the browser is redirected away to a client/default page and no Date/User/Order No/Payment Method table is rendered. |
| 3 | From the same session, request /get-paymentlog (the DataTables JSON source). | No payment log JSON data is returned — the response is a redirect away from the endpoint (or 401 Unauthorized for an AJAX request), not a payload of log rows. |
| 4 | Navigate directly to /settings/system. | The System settings page is not shown and the "Auto Renewal" global toggle is not rendered; the browser is redirected away. |
| 5 | From the same session, send PATCH /settings/system with autorenewal_status=1 and the CSRF token, then log in as the QA_ADMIN_ admin and open /settings/system. | The customer's request did not save anything: the admin sees the "Auto Renewal" toggle in the position it had before the attempt. |

</details>

<details><summary><code>TR0996</code> Disable auto renewal JSON endpoint contract and CSRF handling</summary>

**Precondition:** The QA_CLIENT_ customer owns an order whose Auto Renewal status is "Active"; the order id is known.

| # | Step | Expected result |
|---|---|---|
| 1 | Log in at /login as the QA_CLIENT_ customer, open /my-order/{id} and copy the page's CSRF token. | The order page renders and the "Auto Renewal" tab shows "Status: Active". |
| 2 | Send POST /renewal-disable with the session cookie, the CSRF token and order_id set to that order id. | HTTP 200 with JSON body {"type":"success","message":"Auto subscription Disabled successfully"} |
| 3 | Reload /my-order/{id} and open the "Auto Renewal" tab. | "Status:" now reads "Inactive" and the toggle is off — the endpoint actually changed the state. |
| 4 | Send POST /renewal-disable again with the session cookie and CSRF token but no order_id field. | The request does not report success: the response is HTTP 500 with a JSON body containing a "result" error message (no "type":"success"). |
| 5 | Send POST /renewal-disable with the session cookie and a valid order_id but omit the CSRF token. | The response is a 302 redirect to /login and the order's auto renewal state is unchanged. |

</details>

<details><summary><code>TR0997</code> Failed auto renewal payment leaves auto renewal disabled until the customer re-enables it</summary>

**Precondition:** Requires an order whose automatic renewal attempt has failed at the gateway. There is no UI action that triggers this — it happens when the subscription renewal cron/webhook processes a renewal whose stored card fails (e.g. auto renewal enabled with a Stripe test card that declines on charge, then the renewal job run). If the executor cannot make a renewal attempt fail, mark this case blocked rather than guessing.

| # | Step | Expected result |
|---|---|---|
| 1 | Confirm the starting state: log in as the QA_CLIENT_ customer, open /my-order/{id} for the order and click the "Auto Renewal" tab. | "Status:" reads "Active" in green with "Payment Gateway" and "Subscription Enabled Date:" rows shown. |
| 2 | Cause the order's automatic renewal attempt to run and fail at the gateway (per the precondition), then reload /my-order/{id} and open the "Auto Renewal" tab. | The toggle is off and "Status:" now reads "Inactive" in red; the "Payment Gateway" and "Subscription Enabled Date:" rows are gone. |
| 3 | Log in as the QA_ADMIN_ admin, open /orders/{id} and expand the "Auto Renewal" section. | "Status:" reads "Inactive" there too and the "Auto Renewal for Future Subscription" switch is off and not clickable. |
| 4 | Open /settings/paymentlog as the admin and search for the order number. | A log row for that order records the failed renewal payment (Status shown as the "Failed" link) for the customer. |
| 5 | Log back in as the QA_CLIENT_ customer, open the "Auto Renewal" tab and click the toggle on. | The "Auto Renewal" gateway-selection modal opens and card details are requested again — auto renewal does not resume by itself and is only re-activated after a fresh successful capture. |

</details>


**Next:** review and approve these in QA Touch, then apply `QA: Test case Approved`.
